src/packet/sym_key_encrypted_session_key.rs

```diff
@@ -232,6 +229,52 @@ impl SymKeyEncryptedSessionKey {
             encrypted_key: Some(encrypted_key),
         })
     }
+
+    pub fn encrypt6<F, R: CryptoRng + Rng>(
+        rng: &mut R,
+        msg_pw: F,
+        session_key: &[u8],
+        s2k: StringToKey,
+        sym_algorithm: SymmetricKeyAlgorithm,
+        aead: AeadAlgorithm,
+    ) -> Result<Self>
+    where
+        F: FnOnce() -> String + Clone,
+    {
+        // Initial key material is the s2k derived key.
+        let ikm = s2k.derive_key(&msg_pw(), sym_algorithm.key_size())?;
+        // No salt is used
+        let salt = None;
+
+        let info = [
+            Tag::SymKeyEncryptedSessionKey.encode(), // packet type
+            0x06,                                    // version
+            sym_algorithm.into(),
+            aead.into(),
+        ];
+
+        let hk = hkdf::Hkdf::<Sha256>::new(salt, &ikm);
+        let mut okm = [0u8; 42];
+        hk.expand(&info, &mut okm).expect("42");
+
+        let mut iv = vec![0; aead.iv_size()];
+        rng.fill_bytes(&mut iv);
+
+        // AEAD encrypt
+        let mut encrypted_key = session_key.to_vec();
+        let auth_tag =
+            aead.encrypt_in_place(&sym_algorithm, &okm, &iv, &info, &mut encrypted_key)?;
+
+        Ok(SymKeyEncryptedSessionKey::V6 {
+            packet_version: Default::default(),
```

Comment:
default is this correct?

Reply:
`PacketVersion` defaults to `New`. That's definitely what we want generate for V6 artifacts (and really, always).

Reply:
I added a bit of documentation in 08e6e9bfceea1df71624340b85c56f308e90b20f to make a better connection between the terminology in RFC 9580 and the `Version` enum.